server/src/main/java/org/elasticsearch/action/search/SearchPhaseController.java

```diff
@@ -643,6 +643,7 @@ static ReducedQueryPhase reducedQueryPhase(
             );
             sortedTopDocs = new SortedTopDocs(rankedDocs, false, null, null, null, 0);
             size = sortedTopDocs.scoreDocs.length;
+            from = 0;
```

Comment:
If this is necessary, we need to add a comment as to why.

Reply:
This one is needed because we use the `from` from the `ReducedQueryPhase` in `SearchPhaseController#getHits` in order to trim down the results. Since pagination has already taken place within the ranker, we don't need any additional operations in the later stages.